### PR TITLE
Use PublicKeyBytes as keys in maps to avoid deserialization

### LINKED ISSUE #9222: [crypto][performance] Avoid pubkey deserialization, various instances
from @andll 
- `Header::author` is a `PublicKey`, and it is frequently deserialized (and we have seen this in the flamegraphs)

from @jonas-lj 
- `Serialize::new` calls `highest_round_number` which iterates through `certificate_id_by_round` which is a `DBMap<(Round, PublicKey), CertificateDigest>`.
- This iterator is actually a `RevIter` which wraps the actual `Iter` which is lazily loaded from the DB.
- The iterator skips to the last key in the map and calls next.
- On `Iter::next`, the current key in the map, which is a tuple `(Round, PublicKey)` is deserialized along with the value.
- This last key is returned, but the PublicKey part is ignored and only the highest round is returned.
Something similar seems to happen in last_round also.

### PR DESCRIPTION
**Pull request by jonas-lj**
## Description 

Use PublicKeyBytes instead of PublicKeys as keys in various maps in Narwhal. There could be more places (eg. `Commitee` or `WorkerCache`) where this switch may also be done, but I've done the minimal amount of change in order to fix #9222.

Closing #9222 
Closing (at least partly) #8930

## Test Plan 

All tests in CI passes.

### CODE REVIEW

**Review by kchalkias — COMMENTED** [review]
So are we going to maintain this solution or replace that with indices cc: @andll @jonas-lj , both work!

**Comment by joyqvq** [comment]
do we plan to keep around `struct AuthorityPublicKeyBytes`?

**Review by andll — APPROVED** [review]
Looks good!

@kchalkias - I think this PR is a good step forward. It is not clear to me if ppl have time to do the work that uses indexes instead, so I would prefer to merge this (rather then wait for indexes to be implemented)

**Review by kchalkias — APPROVED** [review]
I do agree w/ @andll, this is already improving CPU utilization and if we ever have a workstream that replaces this with indexes, it can just happen on top of that. Let's rebase and merge.

File: Cargo.toml [hunk 1]
```diff
@@ -129,8 +129,8 @@ move-prover-boogie-backend = { git = "https://github.com/move-language/move", re
 move-stackless-bytecode = { git = "https://github.com/move-language/move", rev = "8729cf2445696cb29f1e0ca06df7d09cf3c2c55f" }
 move-symbol-pool = { git = "https://github.com/move-language/move", rev = "8729cf2445696cb29f1e0ca06df7d09cf3c2c55f" }
 
-fastcrypto = { git = "https://github.com/MystenLabs/fastcrypto", rev = "9fbe55e9d477eac6a6d13e872e1dee19b0c83f53" }
-fastcrypto-zkp = { git = "https://github.com/MystenLabs/fastcrypto", rev = "9fbe55e9d477eac6a6d13e872e1dee19b0c83f53", package = "fastcrypto-zkp" }
+fastcrypto = { git = "https://github.com/MystenLabs/fastcrypto", rev = "c6ceb7df056a7f76ffbe73cfd8996ddab205ab1c" }
```
Comment (kchalkias) [inline Cargo.toml:132]: thanks!

**Review by kchalkias — APPROVED** [review]
Yes, ready to merge now after CI succeeds, thanks for updating the FC pointer

**Review by akichidis — APPROVED** [review]
Thanks @jonas-lj for this! 💯

**Comment by kchalkias** [comment]
@akichidis are you planning to take it to the next step, getting rid of pubKeyBytes as identifiers and use indexes or we believe this is low priority and insignificant for mainnet?

**Comment by akichidis** [comment]
> @akichidis are you planning to take it to the next step, getting rid of pubKeyBytes as identifiers and use indexes or we believe this is low priority and insignificant for mainnet?

@kchalkias I was actually planning to start working on it immediately so we could have this ready by mainnet to minimise as much as possible the impact of deserialisation - as we still keep doing this for Headers etc. It will require more extensive refactoring so I'll also confirm the other priorities in the team.

**Comment by jonas-lj** [comment]
@akichidis Do you have any idea why the format_test fails? Running `cargo -q run --example generate-format -- print`, I get an error:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: UnknownFormatInContainer("StructTag")', crates/sui-core/src/generate_format.rs:121:35
```

**Pull request merged**